src/Shared/AWS/SemanticConventionVersion.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+namespace OpenTelemetry.AWS;
+
+#pragma warning disable SA1300
+#pragma warning disable CA1707
+
+/// <summary>
+/// <para>
+/// Collection of the Open Telemetry Semantic Conventions supported by the OpenTelemetry.*.AWS libraries.
+/// Can be used to pin the version of Semantic Convention emitted.
+/// </para>
+/// <para>
+/// While these libraries are intended for production use, they rely on several Semantic Conventions that are
+/// still considered Experimental, meaning they may undergo additional changes before becoming Stable.  This can
+/// impact the aggregation and analysis of telemetry signals in environments with multiple applications or microservices.
+/// For example, a microservice using an older version of the Semantic Conventions for Http Attributes may emit
+/// <c>"http.method"</c> with a value of GET, while a different microservice, using a new version of Semantic Convention may instead emit the GET as
+/// <c>"http.request.method"</c>.
+/// </para>
+/// <para>
+/// Future versions the OpenTelemetry.*.AWS libraries will include updates to the Semantic Convention, which may break compatibility with a previous version.
+/// To opt-out of automatic upgrades, you can pin to a specific version:
+/// <code>
+/// <![CDATA[
+///  using (var tracerProvider = Sdk.CreateTracerProviderBuilder()
+///     .AddAWSLambdaConfigurations(opt =>
+///     {
+///         opt.SemanticConventionVersion = SemanticConventionVersion.v1_10_EXPERIMENTAL;
+///     })
+///     .Build()!);
+/// ]]>
+/// </code>
+/// </para>
+/// <para>
+/// For additional details, see:
+/// https://opentelemetry.io/docs/specs/otel/versioning-and-stability/.
+/// </para>
+/// </summary>
+/// <remarks>
+/// Once a Semantic Convention becomes Stable, OpenTelemetry.*.AWS libraries will remain on that version until the
+/// next major version bump.
+/// </remarks>
+public enum SemanticConventionVersion
+{
+    /// <summary>
+    /// Use Experimental Conventions until they become stable and then pin to stable.
+    /// </summary>
+    Latest = 0,
+
+    /// <summary>
+    /// Pin to the specific state of all Semantic Conventions as of the 1.10 release of this library.
+    /// https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.27.0.
+    /// </summary>
+    v1_10_Experimental = 1,
+
+    /// <summary>
+    /// Pin to the specific state of all Semantic Conventions as of the 1.10.1 release of this library.
+    /// https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.29.0.
+    /// </summary>
+    v1_10_1_Experimental = 2,
```

Comment:
Consider using semantic convention versions as a names. IMO it will be less confusing.

Reply:
@Kielek - I debated whether to have the naming track the version of OpenTelemetry.*.AWS or the Semantic Convention releases.  

I landed on OpenTelemetry.*.AWS because I thought this would be more approachable/intuitive for users.  a) They're already seeing the version of the OTel library in nuget and in their csproj and b) the SemanticConvention and OpenTelemetry nuget packages update on an independent cadence, so there will inevitably be a miss match.  Ie, OpenTelemetry.*.AWS release include either 0 changes from SemanticConvention, or might use the newest version which is several versions ahead:

_Hypothetical:_
OpenTelemetry.*.AWS 1.10.0 uses SemanticConvnetion 1.27.0
OpenTelemetry.*.AWS 1.10.1 uses SemanticConvnetion 1.29.0
OpenTelemetry.*.AWS 1.10.2 uses SemanticConvnetion 1.29.0
OpenTelemetry.*.AWS 1.11.0 uses SemanticConvnetion 1.35.0


So **Option 1 (current strategy) - Track OpenTelemetry.*.AWS**
```csharp
public enum SemanticConventionVersion
{
   v1_10_0_Experimental, // tracks 1.27.0
   v1_10_1_Experimental, // tracks 1.29.0
   v1_11_0_Experimental, // tracks 1.35.0
}

```

Or **Option 2  - Track SemanticConvention**
```csharp
public enum SemanticConventionVersion
{
   v1_27_0_Experimental, // added in OpenTelemetry.*.AWS 1.10.0
   v1_29_0_Experimental, // added in OpenTelemetry.*.AWS 1.10.1
   v1_35_0_Experimental, // added in OpenTelemetry.*.AWS 1.11.0
}
```

**Very open to suggestion / comments here.  What would be the best user experience?**

Reply:
I would go with Optuon 2.

It is the common sense con Semantic Covention Version.

Option 1 is using same name for different purposes.

Reply:
Sounds good - I'll code up the change.